src/controllers/powerups.ts

```diff
@@ -0,0 +1,68 @@
+import { Request } from "express";
+import { ObjectID } from "mongodb";
+import { BodyParam, Delete, Get, JsonController, Param, Post, Req } from "routing-controllers";
+import { Repository } from "typeorm";
+import { InjectRepository } from "typeorm-typedi-extensions";
+import { BaseController } from ".";
+import { PowerUp } from "../entity/PowerUp";
+import { User } from "../entity/User";
+
+@JsonController("/powerups")
+export class PowerUpsController extends BaseController {
+  constructor(
+    @InjectRepository(User)
+    users: Repository<User>,
+    @InjectRepository(PowerUp)
+    private readonly powerups: Repository<PowerUp>,
+  ) {
+    super(users);
+  }
+
+  @Get("/")
+  public async getAvailablePowerUps(@Req() request: Request) {
+    const user = await this.getUserFromRequest(request);
+
+    return this.powerups.find({
+      availableFor: user.mode,
+    });
+  }
+
+  @Post("/")
+  public async buyPowerUp(@Req() request: Request, @BodyParam("id") powerupId: string, @BodyParam("quantity") quantity: number = 1) {
+    const user = await this.getUserFromRequest(request);
+    const powerups = await this.powerups.findByIds([
```

Comment:
can we use `findOneOrfail` ?

Reply:
We could, but mongo uses `_id` internally, so this is more concise and easier to follow once again.